### PR TITLE
fully move dropck to mir

### PR DESCRIPTION
**Pull request by lcnr**
r? @oli-obk

### CODE REVIEW

**Comment by lcnr** [comment]
alright
@rustbot ready

File: compiler/rustc_typeck/src/check/dropck.rs [hunk 1]
```diff
@@ -231,18 +229,14 @@ fn ensure_drop_predicates_are_implied_by_item_defn<'tcx>(
 /// This function is not only checking that the dropck obligations are met for
 /// the given type, but it's also currently preventing non-regular recursion in
 /// types from causing stack overflows (dropck_no_diverge_on_nonregular_*.rs).
+///
+/// FIXME: Completely rip out dropck and regionck.
 pub(crate) fn check_drop_obligations<'a, 'tcx>(
```
Comment (oli-obk) [inline compiler/rustc_typeck/src/check/dropck.rs:234]: remove this function and move the fixme elsewhere?

**Comment by lcnr** [comment]
I intend to rip all of this out right after this but don't want to have these behavior changes in that big refactor.

@bors r=oli-obk

**Pull request merged**